lib/consumer/consumers_supervisor.ex

```diff
@@ -19,19 +23,24 @@ defmodule Mississippi.Consumer.ConsumersSupervisor do
     queues_config = init_arg[:queues]
 
     children = [
-      {Registry, [keys: :unique, name: Registry.DataUpdater]},
-      {Registry, [keys: :unique, name: Registry.MessageTracker]},
-      {Registry, [keys: :unique, name: Registry.AMQPDataConsumer]},
-      {Consumer.DataUpdater.Supervisor, message_handler: message_handler},
-      {DynamicSupervisor, strategy: :one_for_one, name: Consumer.MessageTracker.Supervisor},
-      {Consumer.AMQPDataConsumer.Supervisor, queues_config: queues_config}
+      {Cluster.Supervisor, [cluster_topologies(), [name: Mississippi.Consumer.ClusterSupervisor]]},
+      {Registry, [keys: :unique, name: DataUpdater.Registry, members: :auto]},
+      {Registry, [keys: :unique, name: MessageTracker.Registry, members: :auto]},
+      {Registry, [keys: :unique, name: AMQPDataConsumer.Registry, members: :auto]},
+      {DataUpdater.Supervisor, message_handler: message_handler},
+      {DynamicSupervisor, strategy: :one_for_one, name: MessageTracker.Supervisor, members: :auto},
+      {AMQPDataConsumer.Supervisor, queues_config: queues_config}
```

Comment:
The `AMQPDataConsumer.Supervisor` is a plain Elixir `Supervisor` and so it contains no way of managing horde membership. It should be distributed, though, even if its children are statically determined.
A quick and dirty way of doing that would be using `Horde.DynamicSupervisor` for the AMQPDataConsumer supervisor too